modules/caddyhttp/reverseproxy/hosts.go

```diff
@@ -144,7 +144,7 @@ func (u *Upstream) fillDialInfo(r *http.Request) (DialInfo, error) {
 	if u.LookupSRV != "" {
 		// perform DNS lookup for SRV records and choose one
 		srvName := repl.ReplaceAll(u.LookupSRV, "")
-		_, records, err := net.DefaultResolver.LookupSRV(r.Context(), "", "", srvName)
+		_, records, err := net.LookupSRV("", "", srvName)
```

Comment:
Why not use DefaultResolver? That is what `net.LookupSRV` does under the hood, and without that, we lose context support, which is important so we can cancel requests without hanging the server during config reloads or shutdowns.

Reply:
The basic reason is to avoid the error log being "flooded" with canceled requests when the client closes the connection before the response is sent. I was getting tons of "operation was canceled" errors because of the context being passed to the LookupSRV being canceled. Using `net.LookupSRV` will indeed use the `DefaultResolver` under the hood but with a new `context.Background()`.

If there's a better way to handle this, I'm all for it! This was just my dirty short way of "fixing" it on my side! :)

Reply:
Ah, and to avoid getting our access logs reporting 502's due to those

Reply:
Not using the context will leak resources; so we can't accept this. But if you want to suppress just "context canceled" errors, I would check for [that error value directly](https://golang.org/pkg/context/#pkg-variables), like `if err == context.Canceled`.  Does that work?

Reply:
yeah, I tried that but the `net` pkg does not return with `context.Canceled` errors... its basically, "operation was cancelled" - https://golang.org/src/net/net.go#L411

Do you have any other suggestions to fix this one? I agree with you it may leak resources, but with my limited understanding of Go that was the fastest and simpler way I found.

Reply:
Ugh, I see, that's annoying.

Worst case scenario, maybe we do text string comparison.

What about converting to net.Error, would this work? https://golang.org/pkg/net/#Error

That source you linked me to has a TODO. Why don't they just return context.Canceled???? They're literally just changing it to some unexported value which is no help.

Reply:
I don't have any preferred method for solving this... I'm happy with whatever way you may find to not log (except in DEBUG) the context cancelled errors for when client connections are closed. Also, not considering these as `502` is required IMO. We've lost a lot of time chasing our own tails on these 502's :) Thinking we were losing packets between caddy and upstreams when the issue was these misleading client closed connections generating 502's